src/components/ConnectWallet/ConnectWalletDialog.tsx

```diff
@@ -46,18 +58,93 @@ interface ButtonProps {
   walletName?: string
   logoSrc?: string
   caption?: string
+  connection?: Web3Connection
   onClick: () => void
 }
 
-function WalletConnectButton({ walletName, logoSrc, caption, onClick }: ButtonProps) {
+const wcQRUriAtom = atom<string | undefined>(undefined)
+
+function WalletConnectButton({ walletName, logoSrc, caption, connection: wcTileConnection, onClick }: ButtonProps) {
+  const [tileConnector] = wcTileConnection as [WalletConnect, Web3ReactHooks]
```

Comment:
You switch between wcTileConnection and wcConnectionTile - stick with one name. They're in different contexts, so they can be re-used.